build-cran-package.sh

```diff
@@ -50,19 +50,20 @@ cd ${TEMP_R_DIR}
     sed -i.bak -e "s/~~VERSION~~/${LGB_VERSION}/" DESCRIPTION
     sed -i.bak -e "s/~~DATE~~/${CURRENT_DATE}/" DESCRIPTION
 
-    # Remove 'region' and 'endregion' pragmas. This won't change
-    # the correctness of the code. CRAN does not allow you
-    # to use compiler flag '-Wno-unknown-pragmas' or
+    # Remove 'region', 'endregion', and 'warning' pragmas.
+    # This won't change the correctness of the code. CRAN does
+    # not allow you to use compiler flag '-Wno-unknown-pragmas' or
     # pragmas that suppress warnings.
     echo "Removing unknown pragmas in headers"
-    for file in src/include/LightGBM/*.h; do
+    for file in $(find . -regex ".*\.\(h\|cpp\|hpp\)$"); do
       sed \
         -i.bak \
         -e 's/^.*#pragma region.*$//' \
         -e 's/^.*#pragma endregion.*$//' \
+        -e 's/^.*#pragma warning.*$//' \
```

Comment:
from https://github.com/microsoft/LightGBM/pull/3460/checks?check_run_id=1265073334, it looks like this did not work for all cases

![image](https://user-images.githubusercontent.com/7608904/96276376-8540e380-0f98-11eb-8031-fe8e42aca58f.png)

but I'm not sure why because your code looks correct :thinking:

Reply:
I tried `sh build-cran-package.sh` and then running `find . -regex ".*\.\(h\|cpp\|hpp\)$"` from inside `lightgbm_r`...I see `cuda_tree_learner.h` in the list of files. So I don't think it's a problem with the `find` command

But maybe there's a difference between my shell and the one running in this CI job. Could you try using single quotes for the argument to `regex`?

```shell
find . -regex '.*\.\(h\|cpp\|hpp\)$'
```

Maybe that trailing `$` sign is being interpreted as the start of interpretation instead of a literal for the regex? I know it feels like that ISN'T it, but it's the only idea I have right now

Reply:
I changed the quotes, but it looks like our issue is going to be more along the lines of what is posted here: https://stackoverflow.com/questions/5905493/why-isnt-this-regex-working-find-regex-m-h

I'll figure out a more portable way to do that find and make another commit soon. (gotta run to work now)

Reply:
Looks like `find ./lightgbm_r/ -name *.h -o -name *.hpp -o -name *.cpp` will work too, it's just ugly. If you can confirm it works on macOS I'll change the code to use that.

Reply:
This might work too:
```
find -E . -regex '.*\.(h|cpp|hpp)$'
```

Reply:
oooooo this is one of those "some of the unix tools work different on Mac" things. yikes haha

Ok thanks for investigating. I might push some attempts later today, when I can try it out on my Mac.

Reply:
I had a few extra minutes this morning so just made that change and pushed it so we could just see if the checks passed. And it looks like they have.